prelude/main.md

```diff
@@ -0,0 +1,71 @@
+# MiniRust prelude
+
+Across all files in this repository, we assume some definitions to always be in scope.
+
+```rust
+/// All operations are fallible, so they return `Result`.  If they fail, that
+/// means the program caused UB or put the machine to a halt.
+pub type Result<T=()> = std::result::Result<T, TerminationInfo>;
+
+#[non_exhaustive]
+pub enum TerminationInfo {
+  Ub(String),
+  MachineStop(String),
+}
+
+/// Some macros for convenient yeeting (yes this is valid syntax on nightly Rust).
```

Comment:
```suggestion
/// Some macros for convenient yeeting, i.e., return an error from a
/// `Option`/`Result`-returning function.
```
But also, we almost certainly don't want these macros in Coq like that.

Can you define a `yeet!` macro in specr so that these here can become `specr::yeet!(())` and `specr::yeet!(TerminationInfo::Ub(...))` etc? This will probably have to invoke the [FromResidual](https://doc.rust-lang.org/nightly/std/ops/trait.FromResidual.html) trait.